### PR TITLE
switch control lists

### PR DESCRIPTION
**Pull request by carinedengler**
Does this solution with a separate view and the drop-down menu for switching control lists meet your expectations? I passed about two thirds of a day on this ticket so far. I intend to add some tests as soon as you confirm me the proposed solution.

### CODE REVIEW

File: app/main/views/corpus.py [hunk 1]
```diff
@@ -19,6 +19,18 @@
 AUTOCOMPLETE_LIMIT = 20
 
 
+def _get_available():
```
Comment (PonteIneptique) [inline app/main/views/corpus.py:22]: I like this.

File: app/templates/macros/nav_macros.html [hunk 2]
```diff
@@ -186,11 +186,13 @@
       <i class="fa fa-list"></i> Corrections history</a></li>
   <li class="nav-item"><a class="nav-link" href="{{url_for("control_lists_bp.get", control_list_id=corpus.control_lists_id)}}">
       <i class="fa fa-cogs"></i> Control List</a></li>
+  <li class="nav-item"><a class="nav-link" href="{{url_for("main.control_list_switch", corpus_id=corpus.id)}}">
```
Comment (PonteIneptique) [inline app/templates/macros/nav_macros.html:189]: I think the way it is right now the link would be viewable by *not owners* which should be avoided.

File: app/templates/main/control_list_switch.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{% extends 'layouts/base.html' %}
+{% import 'macros/form_macros.html' as f %}
+
+{% block content %}
```
Comment (PonteIneptique) [inline app/templates/main/control_list_switch.html:4]: I think you are missing the `<form>`

**Review by PonteIneptique — DISMISSED** [review]
Great. As for tests, the École is actually closed, I am just really bad at not working during my vacations. Vincent Jolivet is the budget decider, I am just the original coder / manager of this repository :)

**Comment by carinedengler** [comment]
> Great. As for tests, the École is actually closed, I am just really bad at not working during my vacations. 

I noticed, I started to feel really bad when pushing something :-) I actually didn't expect any replies before the 24th, so please don't feel obliged to comment on my work until then.

**Comment by PonteIneptique** [comment]
I am really bad when it comes to one of my baby software :)

**Comment by PonteIneptique** [comment]
I saw that the code does not check if the user is the owner of the Corpus before switching the control list :) I don't know where to put it in the review.

File: app/templates/macros/nav_macros.html (hunk 2 above)
Comment (MrGecko) [reply app/templates/macros/nav_macros.html:189]: The view itself should not be accessible either.

**Review by MrGecko — CHANGES_REQUESTED** [review]
Need to prevent _non owners or admins_  to 
1) view the new link
2) access the new view

**Comment by carinedengler** [comment]
> Need to prevent _non owners or admins_ to

just to confirm, do you mean that _non admins_ should not be able to view the link/the new view or that _admins_ should not be able to view it?

**Comment by PonteIneptique** [comment]
I dive in: I think website admin should be able to see it. The view and link should be visible to admin AND owners.

**Comment by carinedengler** [comment]
I pushed an update to make the link/view only accessible to admins/owners of the corpus

**Comment by carinedengler** [comment]
sorry, I didn't notice that the changes had not been included, they have been now

**Comment by PonteIneptique** [comment]
Looking at the proposal, I think this ties also with the issue in #202 , if users connected to this corpus are switched, they won't be switch to the new control list.

**Comment by carinedengler** [comment]
I will look into it

**Comment by MrGecko** [comment]
The #202 has been merged, you should be able to resume your work on this issue

**Comment by carinedengler** [comment]
I integrated managing access to the new/old control lists for all users (it turned out that I couldn't use @FrFerry function for it since it is implicitely for the same control list)

**Comment by MrGecko** [comment]
@PonteIneptique should admin/or private list owners be able to switch a corpus to a private list (if they have the right to access it), or should users only be able to switch to public lists?

**Comment by PonteIneptique** [comment]
You should be able to switch to a private list.

Le jeu. 22 oct. 2020 à 9:48 AM, Julien Pilla <notifications@github.com> a
écrit :

**Comment by MrGecko** [comment]
> You should be able to switch to a private list.

Then i'm not sure I understand the following warning note:
https://github.com/hipster-philology/pyrrha/blob/e9ff1fda394e588654021652154a776322a875c5/app/models/control_lists.py#L114-L120

Indeed ```ControlLists.get_available``` can fetch private lists, but only if you are a member of this list: being an admin does not grant you the right to switch to a control list if you are not explicitly part of it.

**Comment by PonteIneptique** [comment]
Sorry. I'll reformulate. You should be as owner able to switch to private
list. Admin should not.

Le jeu. 22 oct. 2020 à 10:15 AM, Julien Pilla <notifications@github.com> a
écrit :

**Comment by MrGecko** [comment]
Looks good to me then!

@carinedengler I think it woud be nice if the name of the corpus was printed above the form; even better if the current control list was preselected (or just printed next to the corpus name).

**Comment by carinedengler** [comment]
> Looks good to me then!
> 
> @carinedengler I think it woud be nice if the name of the corpus was printed above the form; even better if the current control list was preselected (or just printed next to the corpus name).

I added a line with the current control list name and the current control list is now preselected

**Review by MrGecko — APPROVED** [review]

**Pull request merged**